README.txt
w3.uib.no
============

This repository contains the code needed to run the Drupal based UiB web
application, also known as <http://www.uib.no>.

We use git submodules here, so after the initial clone you need to run:

    git submodule update --init

to grab the rest of the code.

To set up a test site, configure Apache to serve the 'drupal' directory at the
root of some virtual host.

Then set up the Drupal site by invoking:

    bin/site-init w3.uib.example.com
    bin/site-install

The 'doc' directory contains further information you might want to read first.
The 'doc/development-setup.md' explains the details on how to set up your
development environment.  An up-to-date HTML version of the docs should always
be available at <http://phpdoc.devapp.uib.no/w3/> (only available on certain
networks within UiB).


doc/release-notes.md
# Release 125 _2019-10-17_
- Endret filterskjemaet på uib.no/studier/alfabetisk slik at det alltid er ekspandert #22872
- Endret engelsk hjelpetekst for telefonfelt på personsidene #22879
- Lagt til lenke til side med informasjon om personvern nederst på sidene #22843
- Deaktivert dra-og-slipp-funksjon for tekstfelt tilknyttet bilder i brødtekst #22863

# Release 124 _2019-09-26_
- Sperret muligheten for opplasting av filer via Video-fanen i filbehandleren #22769
- Rettet feil i forbindelse med oppretting av nye områder #22800
- Rettet feil i forbindelse med sidenes universelle utforming #22801

# Release 123 _2019-09-12_
- Begrenset tillatte filtyper, samt fjernet unødvendig steg på innsetting av inline-bilder i brødtekstfelt #22706 #22768
- Fjernet "Oversettelse" som dukket opp ved redigering av eksisterende bildefiler #22735
- Oppdatert engelske hjelpetekster i editor #22752
- Rettet problem med engelsk språkversjon av utvekslingsavtaler #22757
- Endret tittel på fanen for opplasting av video fra "Web" til Video" (i engelsk editor) #22720

# Release 122 _2019-09-05_
- Endret grensesnittet for bildeopplasting for å oppfylle GDPR-krav om samtykke fra avbildede personer #21795 #22729 #22734
- Endret data fra tidlig utgave av undervisningsspråk på sider for studieprogram #22658
- Lagt til rutine som månedlig fjerner ubrukte kompetansefelt #22684

# Release 121 _2019-08-15_
- Rettet feil der menyen i mal for satsningsområder forsvant i portrettvisning ved bruk av nettleseren Edge #22538
- Rettet feil der tilknyttede personer i artikler i visse tilfeller overlappet bunnmenyen #22624
- Rettet feil i forbindelse med sidenes universelle utforming #22570
- Justert visningen av introvideo og introtekst på områdesider #22541
- Justert krediteringstekstene til bilder og videoer #22619
- Justert visningen av forsideikonene på middels brede skjermer (nettbrett) #22581
- Endret standard-verdi på LinkedIn-lenke #22539
- Endret lenke til side med mer eksamensinformasjon på engelske emnesider #22653
- Flyttet søke-knapp på sider for EVU-kurs #22621
- Oppdatert dato i rutine som fjerner inaktive innholdsprodusenter #22637

# Release 120 _2019-07-04_
- Rettet feil der listen over visningsalternativer for bilde og video alltid ble nullstilt i mal for feature-artikkel #22510
- Rettet feil der menyen ikke ble vist i nettleseren Edge i mal for satsningsområder #22476
- Rettet feil der menyen ble vist på kartsider i mal for satsningsområder #22535
- Rettet feil der meny-knappene fikk ukorrekt utseende i mal for satsningsområder #22495
- Rettet feil i forbindelse med sidenes universelle utforming #22522

# Release 119 _2019-06-20_
- Gjort endringer på feltet "Ny student" på studieprogramsidene (endret utseende, lagt til datotest og støtte for visning av video) #22464 #22472
- Rettet feil i forbindelse med sidenes universelle utforming #22449 #22461

# Release 118 _2019-06-06_
- Rettet feil i forbindelse med sidenes universelle utforming #22450
- Fjernet lenke fra infoside-overskrift i innholdslister #22451

# Release 117 _2019-05-23_
- Innlogging via https://uib.no/user eller https://uib.no/login sender nå bruker til norsk editor #21652
- Utbedret brødsmulestiene ytterligere #22346
- Fjernet systemadvarsler #22369
- Flyttet "Send"-knapp i 'Media Module' på berøringsskjerm-enheter #22268
- Fikset feil i javascript som førte til at chatboten Hubro på HF-området ikke fungerte #22409
- Oppdatering av diverse Drupal-moduler #22403 #22404 #22405

# Release 116 _2019-05-09_
- Forbedret listen over visningsalternativer for bilde og video i redigeringsgrensesnittet #22367
- Åpnet opp for at også vanlige innholdsprodusenter får tilgang til URL-felt for Google Maps i artikler #22349
- Lagt til "Litterær salong" som ny arrangementstype #22355
- Fjernet feilmeldinger som ble vist på lokale utviklingsinstanser #22347
- Sikkerhetsoppdatert Drupal-kjernen #22365

# Release 115 _2019-04-25_
- Endret brødsmulestiene (bl.a. fjernet siste ledd) #22127
- Lagt til støtte for kobling til PhilPeople på personsider #22270
- Sikkerhetsoppdatert Drupal-kjernen samt installert en modul-patch #22289 #22290

# Release 114 _2019-04-11_
- Fjernet malen "Practical information" (visning av faktaboks i brødtekst) fra publiseringsverktøylinjen på studieprogram der ny mal er brukt, samt fjernet den inaktive koden (* studyinfo *) maskinelt fra teksten #22224
- Lagt til redigerbart felt for "Tuition" i ny studieprogrammal #22223
- Lagt til rette for at teksten "Meir informasjon under" kun blir vist når det faktisk finnes vitnesbyrd på studieprogramsider basert på ny mal #22233
- Rettet feil der riktig tekstformat ikke var satt som standard i brødtekstfeltet på studieprogramsider basert på ny mal #22243
- Rettet feil der visse emnenavn ikke ble vist i studieløpstabeller #22241
- Rettet feil i forbindelse med sidenes universelle utforming #22237
- Rettet feil der undermenyen til globalmenypunktet "Om UiB" ikke kunne åpnes på mobil #22236
- Rettet feil der den nedre områdemenyen ikke kunne åpnes på mobil #22265
- Endret rutinen for visning av kontaktinformasjon på studieprogramsider basert på gammel mal #22259
- Endret utseende på krediteringstekst for videoer #22227
- Fjernet strek under feltet for områdetittel dersom meny ikke finnes #22262
- Justert luften mellom elementene i kolofonfeltet på studieprogramsider basert på ny mal #22266
- Oppdatert Drupal-modul #22269

# Release 113 _2019-03-28_
- Endret hvilke felt som vises i faktalinjen på studieprogramsider basert på ny mal #22129 #21877
- Endret kolofonfeltet på studieprogramsider basert på ny mal #22130
- Fjernet områdemenyen fra studieprogramsider basert på ny mal #22168
- Fjernet en del felt fra visning på studieprogramsider basert på ny mal #22190
- Lagt til rette for at de aller fleste nettlesere nå husker hvilken fane som var valgt når man går tilbake til siden uib.no/studier #22140
- Lagt til fast tekst før vitnesbyrd, video og bilde på studieprogramsider basert på ny mal #22131

modules/uib_tkview/app.js
Ember.$.getJSON('https://tk.app.uib.no/api/v1/service/all', function(data) {
  var tkdata = data;

  // configuration
  var extended_view = false;

  var m;
  if (m = document.URL.match(/\?([^#]+)/)) {
    var q = m[1];
    q = q.split('&');
    for (var i=0; i < q.length; i++) {
      if (q[i] == 'ex')
        extended_view = true;
    }
  }

  var pretty_values = {
    criticality: {
      1: 'kritisk',
      2: 'viktig',
      3: 'normal',
      4: 'liten'
    },
    availability: {
      247: 'alltid',
      'arbeidstid': '08:00–15:45 man–fre',
      'vakt': '07:00-22:00 man-fre (10:00-18:00 lør–søn)'
    },
    service_supplier: {
      'ita': 'IT-avdelingen',
      'intern': 'annen intern enhet',
      'ekstern': 'ekstern',
    },
    service_type: {
      'cfs': 'tilbudt',
      'ss': 'understøttende',
    },
  };

  var pretty = function(obj, attr) {
    v = obj[attr];
    if (pretty_values[attr]) {
      return pretty_values[attr][v] || v;
    }
    return v;
  }

  // filter and prepare the service records for display
  filtered_service = {};
  for (var k in tkdata.service) {
    var s = tkdata.service[k]

    if (!extended_view) {
      if (s.info.service_type != 'cfs')
        continue;
      if (s.info.service_state != 'prod')
        continue;
      if (!extended_view && s.info.edit_state != 'accepted')
        continue;
    }

    s.id = k;
    filtered_service[k] = s;

    s.infolinks = [];
    if (s.link.doc) {
      s.infolinks.push({'title': 'Dokumentasjon av tjenesten', 'icon': 'fa-info-circle', 'link': s.link.doc});
    }
    if (extended_view && s.info.cmdb_id) {
      s.infolinks.push({'title': 'CMDB #' + s.info.cmdb_id, 'icon': 'fa-database', 'link': 'https://bs.uib.no/?module=cmdb&action=viewt&tjeneste=' + s.info.cmdb_id});
    }
    if (extended_view && s.link.href_tk) {
      s.infolinks.push({'title': 'tk.app.uib.no', 'icon': 'fa-file-text-o', 'link': s.link.href_tk});
    }
    if (extended_view && s.link.href_tk) {
      s.infolinks.push({'title': 'Rediger tjenenesten på tk.app.uib.no', 'icon': 'fa-cog', 'link': s.link.href_tk + "/edit"});
    }

    s.facts = []
    if (extended_view && s.info.criticality) {
      s.facts.push({'title': 'Kritikalitet', 'value': pretty(s.info, 'criticality')});
    }
    if (s.info.availability) {
      s.facts.push({'title': 'Tilgjengelighet', 'value': pretty(s.info, 'availability')});
    }
    if (s.info.cost) {
      s.facts.push({'title': 'Kostnad', 'value': s.info.cost});
    }
    if (s.info.service_status) {
      s.facts.push({'title': 'Driftsstatus', 'value': s.info.service_status});
    }
    if (extended_view && s.info.service_type) {
      s.facts.push({'title': 'Tjenestetype', 'value': pretty(s.info, 'service_type')});
    }
    if (extended_view && s.info.service_state) {
      s.facts.push({'title': 'Livssyklus', 'value': pretty(s.info, 'service_state')});
    }
    if (s.info.service_supplier) {
      s.facts.push({'title': 'Driftsleverandør', 'value': pretty(s.info, 'service_supplier')});
    }
    if (s.meta.changed) {
      s.facts.push({'title': 'Oppdatert', 'value': s.meta.changed.substr(0,10)});
    }
    if (extended_view && s.info.cmdb_id) {
      s.facts.push({'title': 'CMDB#', 'value': s.info.cmdb_id});
    }

    if (s.related.support_contact) {
      s.related.support_contact.info = tkdata.support[s.related.support_contact.id];
    }

    if (s.role && s.role.owner) {
      s.role.owner_name = tkdata['user'][s.role.owner];
      s.role.owner_link = 'http://www.uib.no/user/uib/' + s.role.owner;
    }
  }
  tkdata.service = filtered_service

  var categories = [];
  for (var k in tkdata['class']) {
    c = tkdata['class'][k];
    if (c.icon) {
      c.icon = 'fa-' + c.icon;
    }
    else {
      c.icon = 'fa-star';
    }
    c.services = [];
    c.promoted_services = [];

    categories.push(c);
  }
  categories.sort(function(a, b) {
    return a.weight - b.weight;
  });

  for (var k in tkdata.service) {
    s = tkdata.service[k]
    s.categories = [];
    promoted = s.flag && s.flag.search(/promote/ >= 0);
    if ('classification' in s) {
      classification = s['classification'].split(' ');
      for (var i=0; i < classification.length; i++) {
        tkdata['class'][classification[i]].services.push(s);
        s.categories.push(tkdata['class'][classification[i]]);
        if (promoted && tkdata['class'][classification[i]].promoted_services.length < 5) {
          tkdata['class'][classification[i]].promoted_services.push(s);
        }
      }
    }
  }

  var App = Ember.Application.create({
    rootElement: '#ember-app',
  });
  App.Router.map(function() {
    this.resource('about');
    this.resource('services', function() {
      this.resource('service', { path: ':service_id' });
    });
    this.resource('categories', function() {
      this.resource('category', { path: ':category_id' });
    });
    // put your routes here
  });

  App.IndexRoute = Ember.Route.extend({
    model: function() {
      cat = tkdata['class']
      cat_list = []
      for (var i = 0; i < categories.length; i++) {
        if (categories[i].promoted_services.length && categories[i].type != 'service_roles')
          cat_list.push(categories[i]);
      }
      return cat_list
    },
  });

  App.ServicesRoute = Ember.Route.extend({
    model: function() {
      lst = [];
      for (var k in tkdata.service) {
        var s = tkdata.service[k];
        lst.push(s);
      }
      lst.sort(function(a, b) {
        a = a.title;
        b = b.title;
        return a < b ? -1 : a > b ? 1 : 0;
      });
      return lst;
    }
  });

  App.ServicesController = Ember.Controller.extend({
    filteredServices: function() {
      try {
        needle = new RegExp(this.get('searchFor'), 'i');
        return this.get('model').filter(function (o) {
          return o.title.match(needle);
        });
      }
      catch(err) {
        // regexp construction might fail
        return this.get('model');
      }
    }.property('model', 'searchFor'),
  });

  App.ServiceRoute = Ember.Route.extend({
    model: function(params) {
      return tkdata.service[params.service_id];
    }
  });

  App.CategoriesRoute = Ember.Route.extend({
    model: function() {
      return categories;
    }
  });

  App.CategoryRoute = Ember.Route.extend({
    model: function(params) {
      return tkdata['class'][params.category_id];
    }
  });

});
